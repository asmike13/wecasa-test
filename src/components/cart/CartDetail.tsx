import { ICartItem } from 'src/redux/reducers/cartSlice'

export interface ICartDetailProps {
	items: ICartItem[]
	onRemoveOne: (item: ICartItem) => void
	onAddOne: (item: ICartItem) => void
	onRemoveAll: (item: ICartItem) => void
}

const CartDetail = ({
	items,
	onRemoveOne,
	onAddOne,
	onRemoveAll,
}: ICartDetailProps) => {
	return (
		<div className='cart-detail'>
			{items.map((item) => (
				<div className='cart-detail-item' key={item.prestation.reference}>
					<span>
						{`${item.prestation.title} ${item.category.title}`}
					</span>
					<div className='cart-detail-item-update'>
						<button onClick={() => onRemoveOne(item)}>-1</button>
						{`x ${item.prestation.quantity}`}
						<button onClick={() => onAddOne(item)}>+1</button>
						<button onClick={() => onRemoveAll(item)}>Remove</button>
					</div>
				</div>
			))}
		</div>
	)
}

export default CartDetail
