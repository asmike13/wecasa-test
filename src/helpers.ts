export const timeFormat = (timeInMinutes: number) => {
	const hours = Math.floor(timeInMinutes / 60)
	const minutes = timeInMinutes % 60

	const displayHours = hours > 0 ? `${hours}H` : ''
	const displayMinutes = minutes > 0 ? `${minutes}min` : ''

	return `${displayHours} ${displayMinutes}`
}

export const priceFormat = (priceInCents: number) => {
	const price = Math.floor(priceInCents / 100)
	const cents = priceInCents % 100

	const displayPrice = price > 0 ? price : ''
	const displayCents = cents > 0 ? `.${cents}` : ''

	return `${displayPrice}${displayCents}â‚¬`
}
