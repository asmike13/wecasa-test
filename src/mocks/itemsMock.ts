export const itemMock = [{
	key: 'pReference-cReference',
	prestation: {
		reference: 'pReference',
		title: 'pTitle',
		duration: 10,
		price: 10,
		quantity: 10,
	},
	category: {
		reference: 'cReference',
		title: 'cTitle',
	},
}, {
	key: 'pReference2-cReference2',
	prestation: {
		reference: 'pReference2',
		title: 'pTitle2',
		duration: 103,
		price: 13,
		quantity: 3,
	},
	category: {
		reference: 'cReference2',
		title: 'cTitle2',
	},
}]
