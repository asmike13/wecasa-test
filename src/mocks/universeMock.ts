export const universeMock = {
	reference: 'haircut',
	title: 'Coiffure',
	categories: [{
		reference: 'man',
		title: 'Homme',
		prestations: [{
			reference: 'man_shampoo',
			title: 'Shampoing',
			duration: 10,
			price: 400,
		},
		{
			reference: 'man_haircut',
			title: 'Coupe homme',
			duration: 30,
			price: 2990,
		},
		{
			reference: 'man_color',
			title: 'Couleur',
			duration: 45,
			price: 6090,
		},
		{
			reference: 'beard_trim',
			title: 'Taille de la barbe',
			duration: 15,
			price: 2000,
		}],
	},
	{
		reference: 'woman',
		title: 'Femme',
		prestations: [{
			reference: 'woman_shampoo',
			title: 'Shampoing',
			duration: 10,
			price: 400,
		},
		{
			reference: 'woman_haircare',
			title: 'Soin profond',
			duration: 10,
			price: 1000,
		},
		{
			reference: 'woman_haircut',
			title: 'Coupe cheveux courts et mi-longs',
			duration: 30,
			price: 3290,
		},
		{
			reference: 'creative_haircut',
			title: 'Coupe cheveux longs',
			duration: 45,
			price: 4290,
		},
		{
			reference: 'brushing',
			title: 'Brushing (courts et mi-longs)',
			duration: 20,
			price: 3290,
		},
		{
			reference: 'brushing_long_hair',
			title: 'Brushing (longs)',
			duration: 40,
			price: 4290,
		},
		{
			reference: 'woman_color',
			title: 'Couleur (racines)',
			duration: 45,
			price: 6090,
		},
		{
			reference: 'woman_fullcolor',
			title: 'Couleur complète',
			duration: 75,
			price: 8090,
		},
		{
			reference: 'tie_and_dye',
			title: 'Ombré hair',
			duration: 90,
			price: 9090,
		},
		{
			reference: 'woman_streaks',
			title: 'Mèches',
			duration: 120,
			price: 8090,
		},
		{
			reference: 'woman_hair_highlighting',
			title: 'Balayage',
			duration: 120,
			price: 8090,
		},
		{
			reference: 'woman_hair_permanent',
			title: 'Permanente',
			duration: 120,
			price: 8090,
		},
		{
			reference: 'haircare_patina',
			title: 'Patine',
			duration: 15,
			price: 1990,
		},
		{
			reference: 'complex_brushing',
			title: 'Brushing élaboré',
			duration: 40,
			price: 4290,
		},
		{
			reference: 'evening_brushing',
			title: 'Coiffure de soirée (chignon, tresses, attaches…) ',
			duration: 45,
			price: 5990,
		},
		{
			reference: 'sophisticated_hairdressing',
			title: 'Coiffure de mariée',
			duration: 90,
			price: 9900,
		},
		{
			reference: 'hairdressing_bride_with_test',
			title: 'Coiffure de mariage avec essai',
			duration: 90,
			price: 21900,
		},
		{
			reference: 'brazilian_smoothing',
			title: 'Lissage brésilien',
			duration: 180,
			price: 21900,
		}],
	},
	{
		reference: 'child',
		title: 'Enfant',
		prestations: [{
			reference: 'child_haircut',
			title: 'Coupe enfant',
			duration: 20,
			price: 1790,
		},
		{
			reference: 'child_hairdressing',
			title: 'Coiffure enfant (chignon, tresse, attache...) ',
			duration: 30,
			price: 3990,
		}],
	}],
}
