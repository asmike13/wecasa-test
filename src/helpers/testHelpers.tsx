import { Provider } from 'react-redux'
import { store } from 'src/store'

export const StoreWrapper = (props: any) => {
	return (
		<Provider store={store}>
			{props.children}
		</Provider>
	)
}
