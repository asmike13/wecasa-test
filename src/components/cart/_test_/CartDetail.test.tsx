import renderer from 'react-test-renderer';
import { StoreWrapper } from 'src/helpers/testHelpers';
import { itemMock } from 'src/mocks/itemsMock';
import CartDetail, { ICartDetailProps } from '../CartDetail';

export const props: ICartDetailProps = {
	items: itemMock,
	onRemoveOne: jest.fn(),
	onAddOne: jest.fn(),
	onRemoveAll: jest.fn(),
}

it('should render CartDetail as snapshot without items', async () => {
	const snapshot = renderer.create(
		<StoreWrapper>
			<CartDetail {...props} items={[]} />
		</StoreWrapper>,
	)

	expect(snapshot).toMatchSnapshot()
})

it('should render CartDetail as snapshot with items', async () => {
	const snapshot = renderer.create(
		<StoreWrapper>
			<CartDetail {...props} />
		</StoreWrapper>,
	)

	expect(snapshot).toMatchSnapshot()
})
