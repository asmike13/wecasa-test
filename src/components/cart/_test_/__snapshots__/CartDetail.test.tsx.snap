// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CartDetail as snapshot with items 1`] = `
<div
  className="cart-detail"
>
  <div
    className="cart-detail-item"
  >
    <span>
      pTitle cTitle
    </span>
    <div
      className="cart-detail-item-update"
    >
      <button
        onClick={[Function]}
      >
        -1
      </button>
      x 10
      <button
        onClick={[Function]}
      >
        +1
      </button>
      <button
        onClick={[Function]}
      >
        Remove
      </button>
    </div>
  </div>
  <div
    className="cart-detail-item"
  >
    <span>
      pTitle2 cTitle2
    </span>
    <div
      className="cart-detail-item-update"
    >
      <button
        onClick={[Function]}
      >
        -1
      </button>
      x 3
      <button
        onClick={[Function]}
      >
        +1
      </button>
      <button
        onClick={[Function]}
      >
        Remove
      </button>
    </div>
  </div>
</div>
`;

exports[`should render CartDetail as snapshot without items 1`] = `
<div
  className="cart-detail"
/>
`;
